networkx/utils/backends.py

```diff
@@ -53,65 +53,122 @@
+        dispatching to work. The arguments to ``convert_from_nx`` are:
+
+            - ``G`` : NetworkX Graph
+            - ``edge_attrs`` : dict, optional
+                Dictionary mapping edge attributes to default values if missing in ``G``.
+                If None, then no edge attributes will be converted and default may be 1.
+            - ``node_attrs``: dict, optional
+                Dictionary mapping node attributes to default values if missing in ``G``.
+                If None, then no node attributes will be converted.
+            - ``preserve_edge_attrs`` : bool
+                Whether to preserve all edge attributes.
+            - ``preserve_node_attrs`` : bool
+                Whether to preserve all node attributes.
+            - ``preserve_graph_attrs`` : bool
+                Whether to preserve all graph attributes.
+            - ``preserve_all_attrs`` : bool
+                Whether to preserve all graph, node, and edge attributes.
+            - ``name`` : str
+                The name of the algorithm.
+            - ``graph_name`` : str
+                The name of the graph argument being converted.
+
+        2. ``can_run``: 
+                If your backend only partially implements an algorithm, you can define
+                a ``can_run(name, args, kwargs)`` function in your ``Dispatcher`` class that
+                returns True or False indicating whether the backend can run the algorithm with
+                the given arguments or not. Instead of a boolean you can also return a string
+                message to inform the user why that algorithm can't be run.
+
+        3. ``should_run``: 
+                A backend may also define ``should_run(name, args, kwargs)``
+                that is similar to ``can_run``, but answers whether the backend *should* be run
+                (converting if necessary). Like ``can_run``, it receives the original arguments
+                so it can decide whether it should be run by inspecting the arguments.
+                ``can_run`` runs before ``should_run``, so ``should_run`` may assume ``can_run``
+                is True. If not implemented by the backend, ``can_run`` and ``should_run`` are
+                assumed to always return True if the backend implements the algorithm.
+
+        4. ``on_start_tests``: 
+                A special ``on_start_tests(items)`` function may be defined by the backend.
+                It will be called with the list of NetworkX tests discovered. Each item
+                is a test object that can be marked as xfail if the backend does not support
+                the test using ``item.add_marker(pytest.mark.xfail(reason=...))``.
+
+2.  Adding entry points
+
+        To be a valid backend that is discoverable by NetworkX, your package must
+        register an `entry-point <https://packaging.python.org/en/latest/specifications/entry-points/#entry-points>`_
```

Comment:
Making these URLs shorter would be helpful when it is easy to do so. Removing any redundant hashtags is one way... but perhaps they aren't redundant. :)

Also, I'd rather avoid the term "valid backend" because I don't think  that is well defined at this point. 
```suggestion
        To be discoverable by NetworkX, your package must register an 
        `entry-point <https://packaging.python.org/en/latest/specifications/entry-points>`_
```